kernel/mman.c

```diff
@@ -789,21 +782,27 @@ static int _mmap(
         uintptr_t start = (uintptr_t)addr;
         uintptr_t end = (uintptr_t)addr + length;
 
-        /* Fail if [addr:length] is not already mapped */
-        if (!(vad = _list_find(mman, start)) || end > _end(vad))
+        /* Fail if [addr:length] is not already mapped and MAP_FIXED is
+         * requested.
+         */
+        if ((vad = _list_find(mman, start)) && end <= _end(vad))
+        {
+            *ptr_out = addr;
+            goto done;
+        }
+        else if (flags & MYST_MAP_FIXED)
         {
             _mman_set_err(
                 mman,
                 "bad addr parameter: "
-                "must be null or part of an existing mapping");
+                "MAP_FIXED request, but or part of an existing mapping");
```

Comment:
Typo?

Reply:
@mikbras, thanks for spotting this. Fixed.